types/knex/index.d.ts

```diff
@@ -180,6 +180,7 @@ declare namespace Knex {
     interface Join {
         (raw: Raw): QueryBuilder;
         (tableName: TableName, clause: (this: JoinClause) => void): QueryBuilder;
+        (tableName: TableName, clause: (join: JoinClause) => void): QueryBuilder;
```

Comment:
Since the library doesn't know which of these types of function you're providing, it must provide `JoinClause` as both the `this` and regular argument? So could this be `clause: (this: JoinClause, join: JoinClause) => void`?

Reply:
I stand corrected, @andy-ms you are 100% right, and the library does exactly that.